### PR TITLE
Feature/zen 21237

### PR DESCRIPTION
**Pull request by stasfilin**
This is output from tests: 
https://gist.github.com/stasfilin/f3406c51cc154bf80bcf18e66d2dbcc1

### CODE REVIEW

File: ZenPacks/zenoss/OpenStack/tests/utils.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#
+# Copyright (C) Zenoss, Inc. 2016, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+import logging
+LOG = logging.getLogger('zen.OpenStack')
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/utils.py]: LOG is not used in this module, it must be removed

File: ZenPacks/zenoss/OpenStack/tests/utils.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2016, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+import logging
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/utils.py]: logging is not used in this module, it must be removed

File: ZenPacks/zenoss/OpenStack/tests/utils.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+LOG = logging.getLogger('zen.OpenStack')
+
+import Globals
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/utils.py]: I don't think we need import Globals in this module

File: ZenPacks/zenoss/OpenStack/tests/utils.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+from Products.ZenUtils.Utils import unused
+
+unused(Globals)
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/utils.py]: Look above for my comment about Globals

File: ZenPacks/zenoss/OpenStack/tests/utils.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    except ImportError as e:
+        print "\n\nERROR: Unable to import crochet: {}".format(e)
+        print "  You must install it: pip install --no-deps crochet\n"
+        raise
+
+    crochet.setup()
+    return crochet
+
+
+class DictToObject(dict):
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/utils.py]: If this is copied from somewhere there must be an original link as a comment

File: ZenPacks/zenoss/OpenStack/tests/utils.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+LOG = logging.getLogger('zen.OpenStack')
+
+import Globals
+
+from Products.ZenUtils.Utils import unused
+
+unused(Globals)
+
+
+def setup_crochet():
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/utils.py]: This is not used

File: ZenPacks/zenoss/OpenStack/tests/test_modeler.py [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+                               'modeldata.json')) as json_file:
+            results = json.load(json_file)
+
+        results = self._dict_to_object(results)
+
+        for data_map in modeler.process(self.d, results, log):
+            self.applyDataMap(self.d, data_map)
+        self._modeled = True
+
+    def _dict_to_object(self, results):
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/test_modeler.py]: move the method to utils.py, use one style within the module, in this case I'm talking about the method's name

File: ZenPacks/zenoss/OpenStack/tests/test_modeler.py [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+        zcml.load_config('configure.zcml', ZenPacks.zenoss.OpenStack)
+
+        self._loadZenossData()
+
+    def _loadZenossData(self):
+        if hasattr(self, '_modeled'):
+            return
+
+        modeler = OpenStackModeler()
+        with open(os.path.join(os.path.dirname(__file__),
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/test_modeler.py]: extract os.path.join(os.path.dirname(__file__) to a variable, pass it as a variable to open method

File: ZenPacks/zenoss/OpenStack/tests/test_modeler.py [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+import ZenPacks.zenoss.OpenStack
+from ZenPacks.zenoss.OpenStack.tests.utils import setup_crochet
+from ZenPacks.zenoss.OpenStack.modeler.plugins.zenoss.OpenStack \
+import OpenStack as OpenStackModeler
+
+from utils import DictToObject
+
+
+unused(Globals)
+crochet = setup_crochet()
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/test_modeler.py]: not used, must be removed

File: ZenPacks/zenoss/OpenStack/tests/test_modeler.py [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+
+from Products.Zuul.catalog.events import IndexingEvent
+from Products.Five import zcml
+
+from Products.DataCollector.ApplyDataMap import ApplyDataMap
+from Products.ZenTestCase.BaseTestCase import BaseTestCase
+from Products.ZenUtils.Utils import unused
+
+import ZenPacks.zenoss.OpenStack
+from ZenPacks.zenoss.OpenStack.tests.utils import setup_crochet
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/test_modeler.py]: remove this

File: ZenPacks/zenoss/OpenStack/tests/test_modeler.py [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+from ZenPacks.zenoss.OpenStack.modeler.plugins.zenoss.OpenStack \
+import OpenStack as OpenStackModeler
+
+from utils import DictToObject
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/test_modeler.py]: from ZenPacks.zenoss.OpenStack.tests.utils import DictToObject

File: ZenPacks/zenoss/OpenStack/tests/test_modeler.py [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+import logging
+import json
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger('zen.OpenStack')
+
+from zope.event import notify
+
+import Globals
+
+from Products.Zuul.catalog.events import IndexingEvent
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/test_modeler.py]: put it after 'from Products.ZenUtils.Utils import unused' line

File: ZenPacks/zenoss/OpenStack/tests/test_modeler.py [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+import os
+import logging
+import json
+logging.basicConfig(level=logging.INFO)
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/test_modeler.py]: no need to keep this in import section, put this before TestModel class

File: ZenPacks/zenoss/OpenStack/tests/test_modeler.py [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+log = logging.getLogger('zen.OpenStack')
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/test_modeler.py]: no need to keep this in import section, put this before TestModel class

File: ZenPacks/zenoss/OpenStack/tests/test_modeler.py [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+from Products.Zuul.catalog.events import IndexingEvent
+from Products.Five import zcml
+
+from Products.DataCollector.ApplyDataMap import ApplyDataMap
+from Products.ZenTestCase.BaseTestCase import BaseTestCase
+from Products.ZenUtils.Utils import unused
+
+import ZenPacks.zenoss.OpenStack
+from ZenPacks.zenoss.OpenStack.tests.utils import setup_crochet
+from ZenPacks.zenoss.OpenStack.modeler.plugins.zenoss.OpenStack \
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/test_modeler.py]: it must look like this:
from ZenPacks.zenoss.OpenStack.modeler.plugins.zenoss.OpenStack \
    import OpenStack as OpenStackModeler

File: ZenPacks/zenoss/OpenStack/tests/utils.py (hunk 1 above)
Comment (stasfilin) [inline ZenPacks/zenoss/OpenStack/tests/utils.py]: @itshane log needs for 
```python

        for data_map in modeler.process(self.d, results, log):
            self.applyDataMap(self.d, data_map)
```

Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/utils.py]: It's in the other module and is not related to this one

File: ZenPacks/zenoss/OpenStack/tests/test_modeler.py [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+
+        modeler = OpenStackModeler()
+        with open(os.path.join(os.path.dirname(__file__),
+                               'data',
+                               'modeldata.json')) as json_file:
+            results = json.load(json_file)
+
+        results = self._dict_to_object(results)
+
+        for data_map in modeler.process(self.d, results, log):
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/test_modeler.py]: rename log to LOG

File: ZenPacks/zenoss/OpenStack/tests/test_modeler.py [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+                               'data/modeldata.json')) as json_file:
+            results = json.load(json_file)
+
+        results = self._dict_to_object(results)
+
+        for data_map in modeler.process(self.d, results, None):
+            self.applyDataMap(self.d, data_map)
+        self._modeled = True
+
+    def _dict_to_object(self, results):
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/test_modeler.py]: As I mentioned earlier, move this method to utils.py, use the style you have in the module, i.e. camelCase name, for instance 'convertDictsToObjects'.

File: ZenPacks/zenoss/OpenStack/tests/test_modeler.py [hunk 18]
```diff
@@ -0,0 +1,113 @@
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2016, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+import os
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/test_modeler.py]: https://gist.github.com/itshane/3c848b18f443647812e18be7d6a9574d

File: ZenPacks/zenoss/OpenStack/tests/test_modeler.py [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+        self.d.index_object()
+
+        notify(IndexingEvent(self.d))
+        self.applyDataMap = ApplyDataMap()._applyDataMap
+
+        zcml.load_config('configure.zcml', ZenPacks.zenoss.OpenStack)
+
+        self._loadZenossData()
+
+    def _loadZenossData(self):
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/test_modeler.py]: the name should be _loadModelData

File: ZenPacks/zenoss/OpenStack/tests/test_modeler.py [hunk 20]
```diff
@@ -0,0 +1,113 @@
...
+        self.d = dc.createInstance('zenoss.OpenStack.testDevice')
+        self.d.setPerformanceMonitor('localhost')
+        self.d.index_object()
+
+        notify(IndexingEvent(self.d))
+        self.applyDataMap = ApplyDataMap()._applyDataMap
+
+        zcml.load_config('configure.zcml', ZenPacks.zenoss.OpenStack)
+
+        self._loadZenossData()
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/test_modeler.py]: the name should be _loadModelData

File: ZenPacks/zenoss/OpenStack/tests/test_modeler.py [hunk 21]
```diff
@@ -0,0 +1,107 @@
...
+        self.assertTrue(self._modeled)
+
+        images = self.d.getDeviceComponents(type='OpenStackServer')
+        self.assertEquals(len(images), 2)
+
+        self.assertEquals('server0aa87c33-aa73-4c02-976b-321f5e2df205',
+                          images[0].id)
+        self.assertEquals('tiny1', images[0].name())
+
+
```
Comment (itshane) [inline ZenPacks/zenoss/OpenStack/tests/test_modeler.py]: extra line

**Pull request merged**